pkg/cloudprovider/gcp/block_store.go

```diff
@@ -148,12 +154,36 @@ func (b *blockStore) CreateSnapshot(volumeID, volumeAZ string, tags map[string]s
 		snapshotName = volumeID[0:63-len(suffix)] + suffix
 	}
 
+	disk, err := b.gce.Disks.Get(b.project, volumeAZ, volumeID).Do()
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+
+	// upstream uses the description field of GCP disks to store a JSON doc containing
+	// tags.
+	//
+	// use the tags in the disk's description (if a valid JSON doc) plus the tags arg
+	// to set the snapshot's description.
+	var snapshotTags map[string]string
+	if err := json.Unmarshal([]byte(disk.Description), &snapshotTags); err != nil {
+		snapshotTags = tags
```

Comment:
Comment: error decoding the description, so just use our tags.

Might want to log an error?